src/quacc/atoms/skzcam.py

```diff
@@ -23,13 +26,751 @@
         ElementStr,
         MRCCInputDict,
         MultiplicityDict,
+        SKZCAMInfo,
         SKZCAMOutput,
+        semicore_frozen_core_dict,
+        valence_frozen_core_dict,
     )
 
 
+element_from_atomic_num_dict = {v: k for k, v in atomic_numbers.items()}
+
+semicore_frozen_core_dict = {
+    element_from_atomic_num_dict[x]: (
+        0
+        if x <= 4
+        else 2
+        if x <= 12
+        else 10
+        if x <= 30
+        else 18
+        if x <= 38
+        else 28
+        if x <= 48
+        else 36
+        if x <= 71
+        else 46
+        if x <= 80
+        else 68
+        if x <= 103
+        else None  # You can choose a default value or None for x > 103
+    )
+    for x in range(1, 104)  # Adjust the range to include up to 103
+    if x
+    in element_from_atomic_num_dict  # Ensure the atomic number exists in the dictionary
+}
+
+valence_frozen_core_dict = {
+    element_from_atomic_num_dict[x]: (
+        0
+        if x <= 2
+        else 2
+        if x <= 10
+        else 10
+        if x <= 18
+        else 18
+        if x <= 30
+        else 28
+        if x <= 36
+        else 36
+        if x <= 48
+        else 46
+        if x <= 54
+        else 54
+        if x <= 70
+        else 68
+        if x <= 80
+        else 78
+        if x <= 86
+        else 86
+        if x <= 102
+        else 100
+        if x <= 103
+        else None  # You can choose a default value or None for x > 103
+    )
+    for x in range(1, 104)  # Adjust the range to include up to 103
+    if x
+    in element_from_atomic_num_dict  # Ensure the atomic number exists in the dictionary
+}
+
+skzcam_cation_cap_ecp = {
+    "Ti": """NewECP
+N_core 0
+  lmax f
+  s 2
+   1      0.860000       9.191690  2
+   2      0.172000       0.008301  2
+  p 2
+   1      0.860000      17.997720  2
+   2      0.172000      -0.032600  2
+  d 2
+   1      1.600000      -9.504310  2
+   2      0.320000      -0.151370  2
+  f 1
+   1      1.000000000    0.000000000 2
+end""",
+    "Mg": """NewECP
+N_core 0
+lmax f
+s 1
+1      1.732000000   14.676000000 2
+p 1
+1      1.115000000    5.175700000 2
+d 1
+1      1.203000000   -1.816000000 2
+f 1
+1      1.000000000    0.000000000 2
+end""",
+}
+
 has_chemshell = find_spec("chemsh") is not None
 
 
+class SKZCAMInputSet:
+    """
+    A class to generate the complete set of SKZCAM inputs for performing direct MRCC and/or ORCA calculations. We define the possibility to have 4 ONIOM layers, each of which has a low-level (ll) and high-level (hl) calculation (see: doi.org/10.1021/jp962071j). It is assumed that the low-level for the subsequent ONIOM level is the high-level for the previous ONIOM level. The deltaCC calculation is also included as an option.
```

Comment:
The `quacc/atoms/skzcam.py` file and the entire `quacc/atoms` module is meant to be fundamental operations on the `Atoms` object independent of any particular code (e.g. ORCA or MRCC). The code here looks good but really should be moved elsewhere.

All input generation is meant to be done as part of the recipes themselves. You can see an example in the `quacc/recipes/orca` directory. There is a `_base.py` file that contains all the behind-the-scenes logic we want to hide from the core recipe logic itself. I don't know exactly how you are planning things out, but something like that is going to be much better than the current location.